evmcc/evmcc.go

```diff
@@ -127,7 +127,8 @@ func (evmcc *EvmChaincode) Invoke(stub shim.ChaincodeStubInterface) pb.Response
 		}
 
 		// Passing the first 8 bytes contract address just created
-		err := eventSink.Flush(string(contractAddr.Bytes()[0:8]))
+		// Hex Encode before flushing to ensure no non utf-8 characters
```

Comment:
add something like "we need to ensure the fabric chaincode stub gets a utf8 string for the event name, as it will be validated as utf8 ....."
multiple times a long the way through fabric.

or some such.

Reply:
The comment below this is not enough? I don't want to riddle the code in large comments. It makes the readability of the code worse in my opinion.

Reply:
comment seems fine. Probably don't even need the first three lines of this block. I think the last three lines of the block explain the reasoning.

we don't really have a good reason here for the argument to flush, so this approaches a "the comment directly restates the operation" which is never a great comment.